src/pe/debug.rs

```diff
@@ -810,18 +810,18 @@ impl<'a> Iterator for POGOEntryIterator<'a> {
             Err(error) => return Some(Err(error.into())),
         };
 
-        if offset > self.data.len() {
+        // Use >= to avoid empty slice, that we want to emit an error early here for
+        // malformed name in a POGO entry.
+        if offset >= self.data.len() {
             return Some(Err(error::Error::Malformed(format!(
-                "Offset {:#x} is too big for containing name field of POGO entry (rva {:#x} and size {:#X})",
-                offset, rva, size
+                "Offset {offset:#x} is too big for containing name field of POGO entry (rva {rva:#x} and size {size:#X})",
             ))));
         }
         let name = match self.data[offset..].iter().position(|&b| b == 0) {
             Some(pos) => {
                 if offset + pos as usize >= self.data.len() {
                     return Some(Err(error::Error::Malformed(format!(
-                        "Null-terminator for POGO entry (rva {:#x} and size {:#X}) found but exceeds iterator buffer",
-                        rva, size
+                        "Null-terminator for POGO entry (rva {rva:#x} and size {size:#X}) found but exceeds iterator buffer",
                     ))));
                 }
                 let name = &self.data[offset..offset + pos + 1];
```

Comment:
i think `let name = self.data.pread(offset, pos + 1).map_err` is clearer, and you can delete the check above. Because you're returning the error will require some handling; probably some combinator version, but a simple match would also suffice. or perhaps this is best?

```rust
let Some(name) = self.data.pread(offset, pos + 1).ok() else {
  return Some(Err(error::Error::Malformed(format!(
                        "Null-terminator for POGO entry (rva {rva:#x} and size {size:#X}) found but exceeds iterator buffer",
                    ))));
}
```

also directly below it `offset + pos + 1` is incremented; if turn `pread` into `gread` it'll do that for you :)